app/models/discourse_activity_pub_authorization.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+class DiscourseActivityPubAuthorization < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :actor, class_name: "DiscourseActivityPubActor"
+
+  scope :active, -> { where("actor_id IS NOT NULL") }
+
+  before_save :ensure_keys, if: :discourse?
+
+  def discourse?
+    auth_type === self.class.auth_types[:discourse]
+  end
+
+  def mastodon?
+    auth_type === self.class.auth_types[:mastodon]
+  end
+
+  def auth_type_name
+    auth_type && self.class.auth_types[auth_type]
+  end
+
+  def self.auth_types
+    @auth_types ||= Enum.new(discourse: 1, mastodon: 2)
+  end
+
+  protected
+
+  def ensure_keys
+    return unless discourse? && private_key.blank? && public_key.blank?
+    keypair = OpenSSL::PKey::RSA.new(2048)
+    self.private_key = keypair.to_pem
+    self.public_key = keypair.public_key.to_pem
+    save!
+  end
+end
+
+# == Schema Information
+#
+# Table name: discourse_activity_pub_authorizations
+#
+#  id          :bigint           not null, primary key
+#  user_id     :integer          not null
+#  actor_id    :integer
+#  domain      :string
+#  auth_type   :integer
+#  token       :string
+#  private_key :text
+#  public_key  :text
```

Comment:
That's an interesting question. Investigating it a bit, I'm not sure if we want to (although I would say that's a preliminary opinion). Firstly, because they're columns we control (they're not storing data from external actors), and secondly because private and public keys can be of varying length, and it's not entirely clear to me that we wouldn't want to change the length of the keys we're storing at some point. See further: https://stackoverflow.com/questions/39068473/what-is-the-maximum-length-of-private-and-public-rsa-keys